### ISSUE TITLE
Docker build not working

### ISSUE THREAD

**Issue by josteinaj** [daisy/pipeline-assembly#166]
```
➜  pipeline-assembly git:(master) git checkout v1.12.0
➜  pipeline-assembly git:(d0c0eb2) docker build . -t daisy-pipeline-1.12 
Sending build context to Docker daemon  1.999MB
Step 1/13 : FROM maven:3.5-jdk-8 as builder
 ---> 985f3637ded4
Step 2/13 : ADD . /usr/src/daisy-pipeline2
 ---> d310c726a234
Step 3/13 : WORKDIR /usr/src/daisy-pipeline2
 ---> Running in c36ec61fec14
Removing intermediate container c36ec61fec14
 ---> c86b760a0bfd
Step 4/13 : RUN curl -fsSL https://raw.github.com/daisy/maven-parents/travis/settings.xml > $HOME/.m2/settings.xml
 ---> Running in ce179e467996
/bin/sh: 1: cannot create /root/.m2/settings.xml: Directory nonexistent
The command '/bin/sh -c curl -fsSL https://raw.github.com/daisy/maven-parents/travis/settings.xml > $HOME/.m2/settings.xml' returned a non-zero code: 2
```

I can make the build successful with this diff:

```
diff --git a/Dockerfile b/Dockerfile
index 2133a1d..9f7c8ee 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -6,8 +6,8 @@
 FROM maven:3.5-jdk-8 as builder
 ADD . /usr/src/daisy-pipeline2
 WORKDIR /usr/src/daisy-pipeline2
-RUN curl -fsSL https://raw.github.com/daisy/maven-parents/travis/settings.xml > $HOME/.m2/settings.xml
-RUN mvn clean package
+RUN mkdir -p $HOME/.m2 && curl -fsSL https://raw.github.com/daisy/maven-parents/travis/settings.xml > $HOME/.m2/settings.xml
+RUN mvn clean package -P assemble-linux-zip
 RUN cd target && unzip assembly-*-linux.zip
 
 # then use the build artifacts to create an image where the pipeline is installed
```

However, the resulting build does not work the same as the published version on Docker Hub, which presumably was built from the same commit (tag v1.12.0).

**Comment by bertfrees** [comment]
Hmm, this is Christian's Dockerfile. I'm using the other one (Dockerfile.without_builder). To build it I run `make docker`.

**Comment by egli** [comment]
Hey, wait a minute :-). The `curl` line is from you @bertfrees (eb5f9942bb894ef2af6ea7ec3f13c15655337da2). TBH I'm not sure what that does.

**Comment by bertfrees** [comment]
Ah yes I might have messed things up in that commit.

Note that the `mvn clean package -P assemble-linux-zip` will not include the CLI and updater. You can see in the Makefile how the profiles are used. Note also that if you use -Passemble-linux-dir you don't need to unzip step anymore.

The reason why the resulting build does not work the same is probably because of the Java version.

**Comment by josteinaj** [comment]
Oh, I see. So it's Travis that builds the image and not Docker Hub?

I'll try with the same mvn command used in the Makefile and see if I can update `Dockerfile` to the correct version.

**Comment by bertfrees** [comment]
Travis builds it but doesn't push it. I do that manually.

**Comment by josteinaj** [comment]
I've pushed a fix for the `Dockerfile` to the epubcheck-4.2.0 branch: https://github.com/daisy/pipeline-assembly/compare/94adf28...b97d31a
